ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/bgp-peer-groups-structured-config-1.yml

```diff
@@ -141,7 +141,7 @@ route_maps:
     - extcommunity soo 192.168.254.111:1 additive
 loopback_interfaces:
 - name: Loopback0
-  description: EVPN_Overlay_Peering
```

Comment:
This guys config:
```yaml
node_type_keys:
  l3leaf:
    ...
    default_evpn_role: client
...
        bgp-peer-groups-structured-config-1:
          ...
          evpn_route_servers: [ bgp-peer-groups-structured-config-2 ]
          mpls_route_reflectors: [ bgp-peer-groups-structured-config-2 ]
          mpls_overlay_role: client
          evpn_gateway:
            evpn_l3:
              enabled: true
```
and the new logic:
```py3
    def overlay_loopback_interface(self, overlay_loopback_description: str = None) -> str:
        if template_path := self.shared_utils.interface_descriptions_templates.get("overlay_loopback_interface"):
            return self._template(template_path, overlay_loopback_description=overlay_loopback_description)

        if overlay_loopback_description is not None:
            return overlay_loopback_description

        if self.shared_utils.mpls_overlay_role in ["server", "client"]:
            return "MPLS_Overlay_peering"

        if self.shared_utils.mpls_lsr is True:
            return "LSR_Router_ID"

        # Covers L2LS
        if self._overlay_routing_protocol == "none":
            return "Router_ID"

        # Note that the current code will render this for HER and others
        return "EVPN_Overlay_Peering"
```
So now we detect `mpls_overlay_role` instead of only looking at `default_mpls_overlay_role` from `node_type_keys`.
So yes I believe this is correct.